### DISCUSSION TITLE
Keeping App Store Connect API Credentials Secure

### DISCUSSION THREAD

**Discussion by robmathers** [fastlane/fastlane#17811] [Q&A]
I'm looking at moving our setup away from an account and towards an API key.

I'm not seeing anything in the docs around actually keeping the API key secure. With the user/password approach, the credentials manager would keep passwords in keychain automatically, but there doesn't seem to be any equivalent for the API key file and associated IDs. Is there a recommended method for keeping the API key encrypted, rather than requiring our dev team to have the file sitting around unencrypted, or rolling our own solution?

**Comment by dinsen** [comment]
Currently there is no solution build in to Fastlane.
But it will be great if Fastlane could provide a solution.

**Comment by pinkisemils** [comment]
We currently are considering a 2 step approach where one host builds the app and another one uploads it, thus no source code will actually be executed on the host that has the API key. 

Since Fastlane would use the API key to generate JWTs, maybe instead of using the key and generate it's own tokens, it could rely on an external service to do so? A big issue for us is that the API key is not scoped, but we'd be OK with giving out JWT's that allow only uploading builds to test flight.
